File: Backend/nuka-pos/src/main/java/com/nuka/nuka_pos/api/organization/Organization.java
package com.nuka.nuka_pos.api.organization;

import com.nuka.nuka_pos.api.organization.enums.BusinessType;
import com.nuka.nuka_pos.api.organization.enums.ReceiptMode;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * Entity representing an Organization in the system.
 */
@Entity
@Table(name = "organizations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Organization {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String code;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false, unique = true)
    private String phone;

    @Column(nullable = false)
    private String country;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BusinessType businessType;

    @Column(nullable = false)
    private Boolean smartInvoicingEnabled;
    @Column(nullable = false)
    private Boolean barcodeEnabled;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReceiptMode receiptMode;

    @Column(nullable = false)
    private String taxNumber;

    @Column()
    private String logoPath;

    @Column()
    private Boolean isVerified;

    @Column()
    private String verificationCode;

    @Column(nullable = false)
    private LocalDateTime createdAt;

}

File: Frontend/Nuka_POS/lib/models/organization.dart
import 'dart:core';
class Organization {
  late final String name;
  final String code;
  late final String email;
  late final String phone;
  final String country;
  late final String businessType;
  late final bool smartInvoicingEnabled;
  late final bool barcodeEnabled;
  late final String receiptMode;
  late final String taxNumber;
  final String logoPath;
  final bool isVerified;
  final String verificationCode;
  final String createdAt;

  Organization({
    required this.name,
    required this.code,
    required this.email,
    required this.phone,
    required this.country,
    required this.businessType,
    required this.smartInvoicingEnabled,
    required this.barcodeEnabled,
    required this.receiptMode,
    required this.taxNumber,
    required this.logoPath,
    required this.isVerified,
    required this.verificationCode,
    required this.createdAt,
  });

  factory Organization.fromJson(Map<String, dynamic> json) {
    return Organization(
      name: json['name'],
      code: json['code'],
      email: json['email'],
      phone: json['phone'],
      country: json['country'],
      businessType: json['businessType'],
      smartInvoicingEnabled: json['smartInvoicingEnabled'],
      barcodeEnabled: json['barcodeEnabled'],
      receiptMode: json['receiptMode'],
      taxNumber: json['taxNumber'],
      logoPath: json['logoPath'],
      isVerified: json['isVerified'],
      verificationCode: json['verificationCode'],
      createdAt: json['createdAt'],
    );
  }


  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'code': code,
      'email': email,
      'phone': phone,
      'country': country,
      'businessType': businessType,
      'smartInvoicingEnabled': smartInvoicingEnabled,
      'barcodeEnabled': barcodeEnabled,
      'receiptMode': receiptMode,
      'taxNumber': taxNumber,
      'logoPath': logoPath,
      'isVerified': isVerified,
      'verificationCode': verificationCode,
      'createdAt': createdAt,
    };
  }
}
